src/react-native-animated-charts/src/charts/linear/ChartLines.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useContext } from 'react';
+import Animated, {
+  useAnimatedStyle,
+  withTiming,
+} from 'react-native-reanimated';
+import { Svg, Line, LineProps } from 'react-native-svg';
+import { useChartData } from '../../helpers/useChartData';
+import withReanimatedFallback from '../../helpers/withReanimatedFallback';
+
+interface ChartLineProps extends LineProps {
+  color?: string;
+  length: number;
+  thickness: number;
+}
+
+function ChartLineFactory(orientation: 'horizontal' | 'vertical') {
+  const isVertical = orientation == 'vertical';
+  return function ChartLine({
+    color = '#000000',
+    thickness = 2,
+    length,
+    ...props
+  }: ChartLineProps) {
+    const { positionX, dotScale, currentPath, height } = useChartData();
+
+    const currentPositionVerticalLineStyle = useAnimatedStyle(
+      () => ({
+        opacity: dotScale.value,
+        transform: [{ translateX: positionX.value }],
+      }),
+      []
+    );
+
+    const openingPositionHorizontalLineStyle = useAnimatedStyle(() => {
+      return {
+        opacity: currentPath.points.length === 0 ? 0 : 1,
+        transform: [
+          {
+            translateY: withTiming(
+              currentPath?.points?.[0].y * height + 10 || 0
+            ),
+          },
+        ],
+      };
+    }, [currentPath]);
+    return (
+      <Animated.View
+        pointerEvents="none"
+        style={[
+          isVertical
+            ? currentPositionVerticalLineStyle
+            : openingPositionHorizontalLineStyle,
+          {
+            height: isVertical ? length + 20 : thickness,
```

Comment:
why 20?

Reply:
I should've used the same comment as in the original code 😄 

```
// temporary fix for clipped chart
```